### PR TITLE
error-handling: wrap API errors coming from bitbox in go

### PR DESCRIPTION
**Pull request by Tomasvrba**

### CODE REVIEW

File: backend/devices/bitbox02/error.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// limitations under the License.
+
+package bitbox02
+
+const (
+	// 100 errors are reserved for errors coming from the device firmware
+	// Difference namespace should be used for local app errors
+
+	// ErrInvalidInput is returned when the request sends and invalid or unexpected input
+	ErrInvalidInput = 101
```
Comment (benma) [inline backend/devices/bitbox02/error.go]: those don't match the numbers in the firmware.

File: backend/devices/bitbox02/handlers/handlers.go [hunk 2]
```diff
@@ -66,11 +67,23 @@ func (handlers *Handlers) Uninit() {
 	handlers.device = nil
 }
 
+func maybeBB02Err(err error, log *logrus.Entry) map[string]interface{} {
+	result := map[string]interface{}{"success": false}
+
+	if BB02Error, ok := errp.Cause(err).(*bitbox02.Error); ok {
```
Comment (benma) [inline backend/devices/bitbox02/handlers/handlers.go]: `bb02Error` (naming convention, local vars are camelCase)

File: backend/devices/bitbox02/handlers/handlers.go [hunk 3]
```diff
@@ -66,11 +67,23 @@ func (handlers *Handlers) Uninit() {
...
+		result["code"] = BB02Error.Code
+		result["message"] = BB02Error.Message
+		log.WithField("bitbox-error", BB02Error.Code).Warning("Received an error from Bitbox")
```
Comment (benma) [inline backend/devices/bitbox02/handlers/handlers.go]: `bitbox02-error`, `BitBox02`

File: backend/devices/bitbox02/error.go (hunk 1 above)
Comment (Tomasvrba) [reply backend/devices/bitbox02/error.go]: Updated in this PR: https://github.com/shiftdevices/firmware_v2/pull/256/files

**Pull request merged**